src/TypeSpec.Extension/Emitter.Csharp/src/lib/confidentLevels.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import {
+    InputDictionaryType,
+    InputListType,
+    InputModelType,
+    InputType,
+    isInputDictionaryType,
+    isInputListType,
+    isInputModelType
+} from "../type/inputType.js";
+
+export function getConfident(
+    type: InputType,
+    visitedModels: Set<InputModelType>
+): boolean {
+    if (isInputModelType(type)) {
+        return getModelConfident(type, visitedModels);
+    } else if (isInputListType(type)) {
+        return getListConfident(type, visitedModels);
+    } else if (isInputDictionaryType(type)) {
+        return getDictionaryConfident(type, visitedModels);
+    }
+
+    return type.IsConfident ?? true; // by default they are confident
+}
+
+function getModelConfident(
+    type: InputModelType,
+    visitedModels: Set<InputModelType>
+): boolean {
+    if (visitedModels.has(type)) return type.IsConfident ?? true; // this means this has been calculated before or being calculated right now, return
```

Comment:
Shall we make sure that, if the model is visited, the IsConfident should be calculated? It may cause problem, we set the true as isConfident if it is not calculated.

Reply:
The fact is that, if the model is visited, the IsConfident is calculated or being calculated right now.
Returning true here is not setting true.

The isConfident is always calculating in this way:
```
isConfident &&= returnedValue;
```
There here we return true is not making that to be true, this is skipping it based on this simply fact:
the expression `isConfident &&= true` will not change the value of `isConfident`.

Reply:
to reduce the possible confusions, I will add some more comments here.